forc-plugins/forc-crypto/src/content.rs

```diff
@@ -0,0 +1,42 @@
+use std::{convert::Infallible, fs::read, path::PathBuf, str::FromStr};
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Content {
+    Path(PathBuf, Vec<u8>),
+    Binary(Vec<u8>),
+}
+
+impl Content {
+    pub fn from_file_or_binary(input: &str) -> Self {
```

Comment:
Can we add a doc comment here? It's not obvious what `input` is supposed to be supplied here. Without reading the internals there is no way to know a `str` with `0x` will be interpreted as a binary. 

Another option would be to split this into 2 functions. 
1. `from_path`
2. `from_binary_str`

Reply:
@JoshuaBatty I think 0f3729898 is a much better elegant solution. Do you that this approach is simpler, more readable and better?

Reply:
Can we add some tests for it?

Reply:
Done in 6c4545c18